nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/clustered/SocketLoadBalancedFlowFileQueue.java

```diff
@@ -151,15 +151,19 @@ public SocketLoadBalancedFlowFileQueue(final String identifier, final Connection
             // that is not the local node identifier. If the Local Node Identifier is not yet known, that's okay. When it becomes known,
             // the queuePartitions array will be recreated with the appropriate partitions.
             final List<QueuePartition> partitionList = new ArrayList<>();
-            partitionList.add(localPartition);
 
             final NodeIdentifier localNodeId = clusterCoordinator.getLocalNodeIdentifier();
             for (final NodeIdentifier nodeId : sortedNodeIdentifiers) {
                 if (nodeId.equals(localNodeId)) {
-                    continue;
+                    partitionList.add(localPartition);
+                } else {
+                    partitionList.add(createRemotePartition(nodeId));
                 }
+            }
 
-                partitionList.add(createRemotePartition(nodeId));
+            // Ensure that our list of queue partitions always contains the local partition.
```

Comment:
Mark, to understand why you make sure the local partition is on the list. it is not assumed that in the iteration on the array sortedNodeIdentifiers the local partition should always be added when the id node is localNodeId?.

Reply:
¿Is there a case in which this conditional is not met?

Reply:
on the other hand, the order would not be altered if it happens that the partition is not in the list for one of the nodes?

Reply:
Yes, `clusterCoordinator.getLocalNodeIdentifier()` can return `null`. In that case, the local partition will never be added in the loop.

Reply:
all clear.